orchestrator-salt/src/main/resources/salt/salt/resize/scripts/resize-storage-volumes.j2

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env bash
+
+source /opt/salt/scripts/format-and-mount-common.sh
+
+LOG_FILE="/var/log/resize-storage-volumes.log"
+
+device_names=$(blkid | grep -i /dev | cut -d ':' -f 1)
+root_disk=$(get_root_disk)
+root_partition_number=$(get_root_disk_partition_number)
+
+for device in $device_names; do
+  # Check if growpart is required
+  if [[ $device =~ $root_disk ]]; then
+    device=$root_disk
+    if sudo growpart -N $device 1 ; then
+      # Check if we're on an NVMe filesystem
+      if [[ -e "$device" && $(readlink -f /dev/xvda) = "$device" ]]
+      then
+        # Rewrite the partition table so that the partition takes up all the space that it can.
+        log $LOG_FILE "resizing disk for non-NVMe filesystem"
+        # Expand the size of the file system.
+        # Check if we're on AL2
+        STR=$(cat /etc/os-release)
+        SUB="VERSION_ID=\"2\""
+        if [[ "$STR" == *"$SUB"* ]]
```

Comment:
If this is AWS specific, I would prefer to check the `NAME` too along with the version. This also implies that on the other platforms and OS-es we should use resize2fs, correct?

Reply:
yes @daszabo on other platforms too we will be using resize. AWS recommended to check only the version of the OS. and to use xfs_growfs if the version is 2. Name verification isn't required